drivers/can/can_mcp2515.c

```diff
@@ -335,8 +335,11 @@ static int mcp2515_set_timing(const struct device *dev,
 	/* CNF1; SJW<7:6> | BRP<5:0> */
 	__ASSERT(timing->prescaler > 0, "Prescaler should be bigger than zero");
 	uint8_t brp = timing->prescaler - 1;
-	const uint8_t sjw = (timing->sjw - 1) << 6;
-	uint8_t cnf1 = sjw | brp;
+	if (timing->sjw > 0) {
+		dev_data->sjw = (timing->sjw - 1) << 6;
```

Comment:
Can you move the assert check into here instead of checking it for a wider range? Then it's still aligned with the datasheet and the 0 / CAN_SJW_NOCHANGE could just be ignored cleanly
```suggestion
		__ASSERT((timing->sjw >= 1) && (timing->sjw <= 4),
		"1 <= SJW <= 4");
		dev_data->sjw = (timing->sjw - 1) << 6;
```